quinn-udp/src/cmsg/mod.rs

```diff
@@ -1,5 +1,138 @@
+use std::{
+    ffi::{c_int, c_uchar},
+    mem, ptr,
+};
+
 #[cfg(unix)]
 #[path = "unix.rs"]
 mod imp;
 
-pub(crate) use imp::{decode, Aligned, Encoder, Iter};
+pub(crate) use imp::Aligned;
+
+// Helper traits for native types for control messages
+pub(crate) trait MsgHdr {
```

Comment:
For this commit, I'd like to split it into two parts, one that extracts the traits from the surrounding code but leaves as much as possible in the same place in `cmsg/unix.rs`, minimizing the diff, and then another that moves the non-Unix specific stuff over to `mod.rs`. This should make it easier to see what is changed.

By prevailing style, the trait definitions should be near the bottom of the module, below the `Encoder` and other types that rely on them.

Within this trait, let's order the methods as `cmsg_firsthdr()`, then `cmsg_nxthdr()`, then `set_control_len()`, then `control_len()` (and use the same order in the impls).

I'm not a big fan of the mix of C naming conventions (`MsgHdr`, `cmsg_firsthdr()`) and Rust naming conventions (`set_control_len()`). Maybe go with Rust naming conventions for the traits? @Ralith thoughts?

Reply:
I think there's some value in clarity from a direct correspondence to C naming, so I prefer e.g. `MsgHdr` over `MessageHeader`. We're not trying to represent the abstract notion of a header here, but a specific POSIX-specified struct, its API, and their direct analogs in Win32.

That said, we should at least be consistent in either always preserving the literal C "words" (i.e. `controllen`), or always delimiting them (e.g. `cmsg_first_hdr`).

I don't feel too strongly either way, though, since suitable documentation should make this clear enough and it's not exactly a far-reaching abstraction.

Reply:
i will work on this part tomorrow

Reply:
I would prefer delimiting the words everywhere.

Reply:
I was also planning to choose this approach.